docs/connector-development/connector-builder-ui/error-handling.md

```diff
@@ -0,0 +1,84 @@
+# Error Handling
+
+Error handlers allow for the connection to decide how to continue fetching data according to the contents of the response from the partner API. Depending on attributes of the response such as status code, text body, or headers, the connection can continue making requests, retry unsuccessful attempts, or fail the sync.
+
+An error handler is made of two parts, "Backoff strategy" and "Response filter". When the conditions of the response filter are met, the connection will proceed with the sync according to behavior specified. See the [Response filter](#response-filter) section for a detailed breakdown of possible response filter actions. In the event of a failed request that needs to be retried, the backoff strategy determines how long the connection should wait before attempting the request again. 
+
+When an error handler is not implemented for a stream, the connection will default to retrying requests that received a 429 and 5XX status code in the response 5 times using a 5-second exponential backoff. This default retry behavior is recommended if an API's documentation does not specify error handling or retry behavior.
+
+Refer to the documentation of the API you are building a connector for to determine how to handle response errors. There can either be a dedicated section listing expected error responses (ex. [Delighted](https://app.delighted.com/docs/api#http-status-codes)) or API endpoints will list their error responses individually (ex. [Intercom](https://developers.intercom.com/intercom-api-reference/reference/listcompaniesforacontact)). There is also typically a section on rate limiting that summarizes how rate limits are communicated in the response and when to retry.
+
+## Backoff strategies
+
+The API documentation will usually cover when to reattempt a failed request that is retryable. This is often through a `429 Too Many Requests` response status code, but it can vary for different APIs. The following backoff strategies are supported in the connector builder:
+* [Constant](#constant)
+* [Exponential](#exponential)
+* [Wait time from header](#wait-time-from-header)
+* [Wait until time from header](#wait-until-time-from-header)
+
+### Constant
+
+When the API documentation recommends that requests be retried after waiting a constant amount of time, the "Constant" backoff strategy should be set on the error handler.
+
+#### Example
+
+The [Intercom API](https://developers.intercom.com/intercom-api-reference/reference/http-responses) is an API that recommends a constant backoff strategy when retrying requests.
+
+### Exponential
+
+The exponential backoff strategy is similar to constant where the connection waits to retry a request based on a numeric value defined on the connector. When the API documentation recommends that requests be retried after waiting an exponentially increasing amount of time, the "Exponential" backoff strategy should be set on the error handler.
```

Comment:
Would it make sense to give the formula here and the backoff times it will produce? It seems a little abstract in the current form.

The example below doesn't really illustrate it because it's not clear what will come after 4 (6? 8?)